tests/test_config.py

```diff
@@ -0,0 +1,132 @@
+"""Test the config library
```

Comment:
I wonder if we should divide our testing into folders, very much like the code is organized. For example; instead of having `tests/test_config.py`, we can have `tests/config/test_base.py`, `tests/config/test_search.py`, etc. The reason I'd like organizing it like that is because the scope is a lot narrower for each test module. So just with a glance at the testing directories, developers know what each test module is for.

Reply:
Agreed, this is the standard that we should follow, ill take it in that direction.